### PR TITLE
Add ability to pass traffic through; add x-parrot-intended-host header

### PR DESCRIPTION
**Pull request by jbergknoff**
This PR adds some configurability, notably the ability to (1) pass traffic through by default, instead of intercepting it, and (2) create exceptions, so that you can capture or allow traffic bound for particular hosts.

The new configuration options are exposed as environment variables so that they can be adjusted without code changes (reducing the coupling between application code and tests). It may make sense for all of the configuration to be done via environment variables, but that can be decided some other time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +34,28 @@ Any outbound HTTP/HTTPS calls made after this point will not reach their intende
 * `Port`: required integer, the port to which requests should be redirected.
 * `Quiet`: optional boolean (default false), suppress logging of requests.
 
+## Configuration
+
+Beyond specifying the host and port of the interceptor, `parrot.js` can be configured without code changes via the following optional environment variables:
+
+* `PARROT_MODE` the behavior for handling requests, either `INTERCEPT` to intercept and forward to a server under your control (the default), or `PASSTHROUGH` to allow all requests through to their intended destinations.
+* `PARROT_HOST_EXCEPTIONS` a comma-delimited string, listing hosts that should do the opposite of the chosen mode.
```
Comment (nimbus154) [inline README.md]: :+1: flexible whitelist or blacklist options

File: parrot.js [hunk 2]
```diff
@@ -20,32 +23,44 @@ function HijackedRequest(MetaOptions)
...
 	{
+		var NewOptions = Options;
 		if (typeof Options === "string")
-			Options = URL.parse(Options);
+			NewOptions = URL.parse(Options);
 
-		if (!MetaOptions.Quiet)
+		var Hostname = NewOptions.hostname || NewOptions.host;
+		var HostIsExceptional = HostExceptions.some(function(Host) { return Host === Hostname; })
+		var RequestShouldPassThrough = (Mode === "PASSTHROUGH") ^ HostIsExceptional;
```
Comment (nimbus154) [inline parrot.js]: :+1: use of xor

File: parrot.js [hunk 3]
```diff
@@ -20,32 +23,44 @@ function HijackedRequest(MetaOptions)
...
 		{
-			console.log
-			(
-				"*** Hijacking HTTP/S request %s //%s%s%s",
-				Options.method,
-				Options.hostname || Options.host,
-				Options.port ? (":" + Options.port) : "",
-				Options.path
-			);
+			if (!MetaOptions.Quiet)
```
Comment (nimbus154) [inline parrot.js]: Any desire to create a wrapper function that checks for `Quiet` flag before logging? The `LogRequest` function shouldn't contain the check, but a wrapper function might avoid the slight bit of logic duplication.

In any case, it's only repeated twice, so not a big deal.

**Comment by nimbus154** [comment]
The header provides a helpful way for the redirect target to resolve path collisions.

I like the flexibility provided by the whitelist and blacklist options.

I don't see any logic errors, and the features seem useful. I'm for merging. :shipit:

File: parrot.js [hunk 4]
```diff
@@ -20,32 +23,44 @@ function HijackedRequest(MetaOptions)
...
 		// Without this, the request may not be intercepted.
-		Options.agent = null;
+		NewOptions.agent = null;
 
 		// Forward as HTTP, even if original request should have been HTTPS.
-		Options.protocol = null;
+		NewOptions.protocol = null;
 
-		return new HTTP.ClientRequest(Options, Callback);
+		return new HTTP.ClientRequest(NewOptions, Callback);
```
Comment (camsjams) [inline parrot.js]: I'd have to dig up the error message, but I recall this line causing errors with newer versions of Node.js

What version has this been tested on?

Comment (jbergknoff) [inline parrot.js]: I think this is the error you've seen: https://github.com/MakerStudios/parrot.js/pull/2. I've tested this on v0.12.9.

File: parrot.js (hunk 3 above)
Comment (jbergknoff) [inline parrot.js]: That's a good point, but I'll leave it like this for now and we can clean it up/refactor at some point in the future.

File: parrot.js (hunk 4 above)
Comment (camsjams) [inline parrot.js]: Ah I see. Great!

**Pull request merged**